docs/guides/advanced/data/4844_data.md

```diff
@@ -0,0 +1,185 @@
+| Trusted setup              | Obtain the [trusted setup][trusted_setup] from Ethereum's KZG ceremony (also known as the secret point 's').                                                                                                                                 |
+| Compute Commitment         | [Calculate the commitment][compute_commitment] `(c = p(s))`.                                                                                                                                                                                 |
+| Versioned Hash:            | Generate a versioned hash (`vh = hash(c)`) which is more flexible and smaller in size than the commitment. (32 bytes vs 48)                                                                                                                  |
+| Choose a Random Point:     | Select a 'random' point `x = hash(hash(data) \|\| vh)` using the [hash of the data and the versioned hash][random_point_selection]. This is ['Fiat-Shamir' heuristic][fiat_shamir]                                                           |
+| Compute Polynomial Value:  | Calculate the value `p(x) = y`.                                                                                                                                                                                                              |
+| Prepare for Circuit Input: | Ensure that `hash(vh \|\| x \|\| y)` is provided as a [public input for the circuit][scheduler_expect_4844], based on the [Schwarz-Zippel lemma][schwarz_zippel] - if 2 polynomials are equal in a random point, there are equal everywhere. |
+
+### Inside the state keeper
+
+| step                         | explanation                                            |
+| ---------------------------- | ------------------------------------------------------ |
+| Collect data                 | Get `c`, `x`, `y`, `blob_proof` from witness generator |
+| Create New Transaction Type: | Form a new blob transaction with a special `0x3` type. |
+
+### Inside the L1 contract
+
+| step                             | explanation                                                                                                                                                                                                              |
+| -------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Receive Inputs:                  | Get `c`, `x`, `y`, and `blob_proof` as inputs from the [state keeper][executor_inputs].                                                                                                                                  |
```

Comment:
What do you mean "from the state keeper"? Interaction with contracts are via transactions